subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -174,9 +177,14 @@ protected ActorFactory getActorFactory() {
         throw new UnsupportedOperationException();
     }
 
-    @Inject
+    @Internal
+    @Deprecated
     protected ClassLoaderCache getClassLoaderCache() {
-        throw new UnsupportedOperationException();
+        DeprecationLogger.deprecateMethod(Test.class, "getClassLoaderCache()")
```

Comment:
if we don't need this inject method, we can just remove it rather than deprecate it.

Reply:
I went with deprecation because it's a breaking change for code extending `Test`.
Are you saying we should make the breaking change?

Reply:
yes, for `@Inject` methods. We do this pretty regularly.